app-website-4/ProductsApi/Auth/TokenProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace ProductsApi.Auth
+{
+    public class TokenProvider
+    {
+        public string GenerateToken(string[] usernameAndPassword)
+        {
+            if (usernameAndPassword[0] == "admin" && usernameAndPassword[1] == "password")
+            {
+                var claimsdata = new[] { new Claim(ClaimTypes.Name, usernameAndPassword[0]) };
+                var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes("This is a secret key"));
```

Comment:
I'd extract a service that returns the secret key as a string (even if it's hard coded in that class).

That makes it clear that this key shouldn't be stored just in source control, but in reality would come from somewhere like a database, environment variables, or some other secure store. I'd even add a comment detailing that on the class.

The last thing we want to do is give people the impression this is a safe way to store these things. (I know you already know this, but others might not 😛)

Reply:
Yeah, that's a fair point, I was going to make this clear in the README but extracting it makes it explicit.